substrate/client/rpc-spec-v2/src/archive/mod.rs

```diff
@@ -33,3 +33,76 @@ pub mod error;
 
 pub use api::ArchiveApiServer;
 pub use archive::Archive;
+
+use serde::{Serialize, Deserialize};
+
+/// The result of an RPC method.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(untagged)]
+pub enum MethodResult {
+	/// Method generated a result.
+	Ok(MethodResultOk),
+	/// Method encountered an error.
+	Err(MethodResultErr),
+}
+
+impl MethodResult {
+	/// Constructs a successful result.
+	pub(crate) fn ok(result: impl Into<String>) -> MethodResult {
+		MethodResult::Ok(MethodResultOk { success: true, value: result.into() })
+	}
+
+	/// Constructs an error result.
+	pub(crate) fn err(error: impl Into<String>) -> MethodResult {
+		MethodResult::Err(MethodResultErr { success: false, error: error.into() })
+	}
+}
+
+/// The successful result of an RPC method.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct MethodResultOk {
+	/// Method was successful.
+	success: bool,
```

Comment:
/cc @lexnv why are we not consistent here with the fields i.e, success private and value public?

I guess this type is not really used by users rather just serialize the response to JSON but we would be better 
to make all public when it's already exposed.